### PR TITLE
Alternative span context implementation

### PR DESCRIPTION
**Pull request by cliedeman**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -109,14 +111,13 @@ export default class OpentracingExtension<TContext extends SpanContext>
     }
 
     const name = info.fieldName || "field";
-    const parentSpan = context.span ? context.span : this.requestSpan;
```
Comment (cliedeman) [inline src/index.ts]: This is actually unrelated but also likely an issue. I used a similar approach in my own extension but this does not work as expected. The child spans end up being nested underneath the parent but they actually execute sequentially so trace timings become distorted.


Query:
```{
{ 
 parent: {
    field1
    field2
  }
}
```

Traces:
```
|--------------parent---------------------|
    |--childField1--||--childField2--|
```

What actually happens
```
|--parent--||--childField1--||--childField2--|
```

**Comment by DanielMSchmidt** [comment]
Thank you for your help. I will take a deeper look later on, but for me to gain certainty that it helps I would like to have a test added that fails with the current implementation. Otherwise I would fear that we reintroduce this bug in future versions

**Comment by DanielMSchmidt** [comment]
I will try to get #49 done, so that we can move forward on this with the backup of actual tests failing

**Comment by DanielMSchmidt** [comment]
Thank you for your help here, I added you into our all contributors section 👍 

Going to close this as we move forward in #83

**Pull request closed without merging**